core/pkg/server/server.go

```diff
@@ -55,9 +60,35 @@ func NewServer(ctx context.Context, addr string, portFile string) (*Server, erro
 
 	s.wg.Add(1)
 	go s.Serve()
+	if pid != 0 {
+		s.wg.Add(1)
+		go s.WatchParentPid(pid)
+	}
 	return s, nil
 }
 
+func (s *Server) loopCheckIfParentGone(pid int) bool {
+	for {
+		select {
+		case <-s.pidwatchChan:
+			return false
+		case <-time.After(100 * time.Millisecond):
+		}
+		parentpid := os.Getppid()
+		if parentpid != pid {
+			return true
+		}
+	}
+}
+
+func (s *Server) WatchParentPid(pid int) {
+	shouldExit := s.loopCheckIfParentGone(pid)
+	if shouldExit {
+		os.Exit(2)
```

Comment:
why 2? "Misuse of Shell Built-in"?

Reply:
changed to 1.. just trying something a bit more unique